lib/bcoin.js

```diff
@@ -102,6 +105,84 @@ class LedgerBcoin {
     });
   }
 
+  /**
+   * Get signatures for transaction
+   * @param {bcoin.TX|bcoin.MTX|Buffer} tx - transaction
+   * @param {bcoin.CoinView|Buffer} view
+   * @param {LedgerTXInput[]} ledgerInputs
+   * @returns {Buffer[]}
+   * @throws {LedgerError}
+   * @throws {AssertionError}
+   */
+
+  async getTransactionSignatures(tx, view, ledgerInputs) {
+    const unlock = await this.txlock.lock();
+
+    try {
+      this.signingTX = true;
+      return await this._getTransactionSignatures(tx, view, ledgerInputs);
+    } finally {
+      this.signingTX = false;
+      unlock();
+    }
+  }
+
+  /**
+   * Sign one input in transaction
+   * @async
+   * @returns {Buffer[]}
+   */
+
+  async _getTransactionSignatures(tx, view, ledgerInputs) {
+    if (Buffer.isBuffer(tx))
+      tx = MTX.fromRaw(tx);
+
+    if (TX.isTX(tx))
+      tx = MTX.fromTX(tx);
+
+    if (Buffer.isBuffer(view))
+      view = CoinView.fromReader(read(view), view);
+
+    assert(MTX.isMTX(tx), 'Can not use non-MTX tx for signing');
+    assert(view instanceof CoinView,
+      'Can not use non-CoinView view for signing');
+
+    const mtx = tx;
+    mtx.view = view;
+
+    const sigstate = SignatureObject.fromOptions({
+      mtx: mtx,
+      ledger: this.ledger,
+      inputs: ledgerInputs
+    });
+
+    sigstate.init();
+
+    await sigstate.collectPubkeys();
```

Comment:
This s where all the ledger communication happens correct? Maybe add a little comment indicating that